src/Simplex_tree/include/gudhi/Simplex_tree/multi_filtrations/Finitely_critical_filtrations.h

```diff
@@ -0,0 +1,1051 @@
+ * i.e., the order \f$\le\f$  in \f$\mathbb R^n\f$ becomes \f$\ge\f$. 
+ *
+ * if `multi_filtration_` contains the points \f$(a_1, a_2, \ldots, a_k) \in (\mathbb R^n)^k\f$,
+ * then a new point \f$x\f$ will be in this filtration if there exists an \f$a_i\f$ such that 
+ * \f$a_i \le x\f$.
+ *
+ */
+template <typename T, bool co=false> class Multi_critical_filtration {
+public:
+  using value_type = T;
+  using OneCritical = One_critical_filtration<T>; // Type of the One critical subtype
+  template <typename value_type>
+  using Base = Multi_critical_filtration<value_type>;
+  std::size_t num_parameters() const { return multi_filtration_.size() == 0 ? 0 : multi_filtration_[0].size(); }
+  Multi_critical_filtration() : multi_filtration_(1)
+  {
+  multi_filtration_[0] = co ? One_critical_filtration<T>::inf() : One_critical_filtration<T>::minus_inf();
+  }; // initialize at 1, with 0 parameter (consistent with below)
+  Multi_critical_filtration(int n)
+      : multi_filtration_({One_critical_filtration<T>(
+            n)}){}; // minus infinity by default
+  Multi_critical_filtration(int n, T value)
+      : multi_filtration_({One_critical_filtration<T>(
+            n, value)}){}; // std::vector<T>(n, value){};
+  Multi_critical_filtration(std::initializer_list<T> init)
+      : multi_filtration_({One_critical_filtration<T>(
+            init)}){}; // : std::vector<T>({std::vector<T>(init)};
+  Multi_critical_filtration(const std::vector<T> &v) : multi_filtration_({v}){};
+  Multi_critical_filtration(std::vector<T> &&v) : multi_filtration_({std::move(v)}){};
+  Multi_critical_filtration(const std::vector<One_critical_filtration<T>> &v) : multi_filtration_(v){};
+  Multi_critical_filtration(std::vector<One_critical_filtration<T>> &&v) : multi_filtration_(std::move(v)){};
+  Multi_critical_filtration(typename std::vector<T>::iterator it_begin,
+                      typename std::vector<T>::iterator it_end)
+      : multi_filtration_(
+            {One_critical_filtration<T>(it_begin, it_end)}){};
+  Multi_critical_filtration(typename std::vector<T>::const_iterator it_begin,
+                      typename std::vector<T>::const_iterator it_end)
+      : multi_filtration_({One_critical_filtration<T>(
+            it_begin, it_end)}){}; // : std::vector<T>(it_begin, it_end){};
+  Multi_critical_filtration(const Multi_critical_filtration<T> &other)
+      : multi_filtration_(other.multi_filtration_){};
+
+  Multi_critical_filtration &operator=(const Multi_critical_filtration<T> &other) {
+    this->multi_filtration_ = other.multi_filtration_;
+    return *this;
+  }
+  inline friend bool
+  operator==(const Multi_critical_filtration<T,co> &self,
+             const Multi_critical_filtration<T,co> &to_compare) {
+    if (self.num_generators() != to_compare.num_generators())
+      return false;
+    auto it = to_compare.begin();
+    for (auto i = 0u; i < self.num_generators(); i++) {
+      if (self[i] != *(it++))
+        return false;
```

Comment:
2 multi-critical lists with the same points in a different order are considered different?